### PR TITLE
[techDebt]: [DEL-3837]: Move delegate related code out of 400-rest

### PR DESCRIPTION
**Pull request by raghuAtWings**
Sanity Shell + HTTP Workflow succeeded:
image

---
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33322)

### CODE REVIEW

**Review by johannesHarness — DISMISSED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33322)* status: :shipit: complete! ✓

**Review by jennyjames21 — APPROVED** [review]
> Reviewed 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33322)* status: :shipit: complete! ✓

File: 260-delegate/src/test/resources/kryo-registrations.txt [hunk 1]
```diff
@@ -810,33 +687,24 @@
...
 7226:software.wings.service.impl.MasterUrlFetchTaskParameter
-7227:software.wings.sm.states.spotinst.SpotInstListenerUpdateStateExecutionData
 7228:software.wings.beans.CyberArkConfig
 7229:software.wings.api.ShellScriptProvisionerOutputElement
 7230:com.google.api.services.logging.v2.model.LogEntry
 7231:com.google.api.services.logging.v2.model.MonitoredResourceMetadata
 7232:com.google.api.services.logging.v2.model.LogEntryOperation
 7233:com.google.api.services.logging.v2.model.MonitoredResource
 7234:com.google.api.services.logging.v2.model.LogEntrySourceLocation
-7235:java.time.Instant
```
Comment (johannesHarness) [inline 260-delegate/src/test/resources/kryo-registrations.txt:829]: Still exists, just moved it to HKryo.

**Comment by johannesHarness** [comment]
@raghuAtWings test failures aren't caused by our change, but the following commit:
https://github.com/harness/harness-core/commit/2e82bc442b8f3f0f1ea15c9b2169c3b04d93a795

I talked to Deepak and he's currently working on a fix for the tests.

**Comment by johannesHarness** [comment]
trigger ti3

**Review by johannesHarness — APPROVED** [review]
> Reviewed 1 of 6 files at r2, 91 of 92 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33322)* status: :shipit: complete! ✓

**Review by jennyjames21 — DISMISSED** [review]
> Reviewed 91 of 92 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33322)* status: :shipit: complete! ✓

**Comment by raghuAtWings** [comment]
*[`400-rest/src/main/java/software/wings/api/ContextElementParamMapperFactory.java` line 88 at r7](https://reviewable.io/reviews/harness/harness-core/33322#-N2Y0ri58em8sJkGgofn:-N2Y0ri68-uckDj3oOnQ:bqebhii) ([raw file](https://github.com/harness/harness-core/blob/44fb83f654fcb05f5d172722b4f0a88a27a49fb3/400-rest/src/main/java/software/wings/api/ContextElementParamMapperFactory.java#L88)):*
> ```Java
>       return new PcfInstanceElementParamMapper((PcfInstanceElement) element);
>     } else if (element instanceof PhaseElement) {
>       return new PhaseElementParamMapper(this.artifactService, this.featureFlagService, (PhaseElement) element);
> ```

why this change?

**Review by raghuAtWings — COMMENTED** [review]
> Reviewed 1 of 6 files at r2, 91 of 92 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33322)* status: :shipit: complete! ✓

**Review by johannesHarness — APPROVED** [review]
> Reviewed 85 of 88 files at r8, 26 of 26 files at r9, 1 of 1 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33322)* status: :shipit: complete! ✓

**Review by johannesHarness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33322)* status: :shipit: complete! ✓

___
*[`400-rest/src/main/java/software/wings/api/ContextElementParamMapperFactory.java` line 88 at r7](https://reviewable.io/reviews/harness/harness-core/33322#-N2Y0ri58em8sJkGgofn:-N2nE3x33m6qvQ09JMFy:b2y0d5k) ([raw file](https://github.com/harness/harness-core/blob/44fb83f654fcb05f5d172722b4f0a88a27a49fb3/400-rest/src/main/java/software/wings/api/ContextElementParamMapperFactory.java#L88)):*
<details><summary><i>Previously, raghuAtWings (Raghvendra Singh) wrote…</i></summary><blockquote>

why this change?
</blockquote></details>

This was previously injected into the `PhaseElement` class.
I just pulled it out in the parammapper (where it's used) and remove it from the bean class.
That allows me to move the `PhaseElement` class to 930+

**Review by puneetsar — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33322)* status: :shipit: complete! ✓

**Pull request merged**